be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -260,10 +266,53 @@ Status LakePersistentIndex::replace(size_t n, const Slice* keys, const IndexValu
     return Status::OK();
 }
 
+void LakePersistentIndex::pick_sstables_for_merge(const PersistentIndexSstableMetaPB& sstable_meta,
+                                                  std::vector<PersistentIndexSstablePB>* sstables,
+                                                  bool* merge_base_level) {
+    // There are two levels in persistent index:
+    //  1) base level. It contains only one sst file.
+    //  2) cumulative level. Sst files that except base level.
+    // And there are two kinds of merge:
+    //  1) base merge. Merge all sst files.
+    //  2) cumulative merge. Only merge cumulative sst files.
+    //
+    // And we use this strategy to decide whether to use base merge or cumulative merge:
+    // 1. When total size of cumulative level sst files reach 1/10 of base level, use base merge.
+    // 2. Otherwise, use cumulative merge.
+    DCHECK(sstable_meta.sstables_size() > 0);
+    int64_t base_level_bytes = 0;
+    int64_t cumulative_level_bytes = 0;
+    std::vector<PersistentIndexSstablePB> cumulative_sstables;
+    for (int i = 0; i < sstable_meta.sstables_size(); i++) {
+        if (i == 0) {
+            base_level_bytes = sstable_meta.sstables(i).filesize();
+        } else {
+            cumulative_level_bytes += sstable_meta.sstables(i).filesize();
+            cumulative_sstables.push_back(sstable_meta.sstables(i));
+        }
+    }
+
+    if (cumulative_level_bytes * 10 < base_level_bytes) {
```

Comment:
would be preferable to make this `10` as a configurable item, like the base-cumulative ratio.

Reply:
Add `config::lake_pk_index_base_cumulative_compaction_ratio`, default is 10.